helper_scripts/stats_helpers.py

```diff
@@ -178,24 +195,40 @@ def get_blocking(self):
         """
         if self.engine_props['num_requests'] == 0:
             blocking_prob = 0
+            bit_rate_blocking_prob = 0
         else:
             blocking_prob = self.blocked_reqs / self.engine_props['num_requests']
+            bit_rate_blocking_prob = self.bit_rate_blocked / self.bit_rate_request
 
         self.stats_props.sim_block_list.append(blocking_prob)
-
+        self.stats_props.sim_bit_rate_block_list.append(bit_rate_blocking_prob)
     def _handle_iter_lists(self, sdn_data: object):
         for stat_key in sdn_data.stat_key_list:
             # TODO: Eventually change this name (sdn_data)
             curr_sdn_data = sdn_data.get_data(key=stat_key)
-
+            if stat_key == 'xt_list':
+                self.stats_props.xt_list.append(mean(curr_sdn_data)) # TODO: double-check
             for i, data in enumerate(curr_sdn_data):
                 if stat_key == 'core_list':
                     self.stats_props.cores_dict[data] += 1
                 elif stat_key == 'modulation_list':
                     bandwidth = sdn_data.bandwidth_list[i]
+                    band = sdn_data.band_list[i]
                     self.stats_props.mods_used_dict[bandwidth][data] += 1
-                elif stat_key == 'xt_list':
-                    self.stats_props.xt_list.append(data)
+                    self.stats_props.mods_used_dict[data][band] += 1
+                    self.stats_props.mods_used_dict[data]['length'][band].append(sdn_data.path_weight)
+                    self.stats_props.mods_used_dict[data]['length']['overall'].append(sdn_data.path_weight)
+                # elif stat_key == 'xt_list':
+                #     self.stats_props.xt_list.append(data) # TODO: double-check
+                elif stat_key == 'start_slot_list':
```

Comment:
I think @kbempah  did something where we could loop over a list of items and access the member in a class with a string. It's probably worth looking into as this scales.